File: app/src/main/java/com/coppel/rhconecta/dev/domain/home/HomeRepository.java
package com.coppel.rhconecta.dev.domain.home;

import com.coppel.rhconecta.dev.business.models.ZendeskResponse;
import com.coppel.rhconecta.dev.domain.common.Either;
import com.coppel.rhconecta.dev.domain.common.UseCase;
import com.coppel.rhconecta.dev.domain.common.failure.Failure;
import com.coppel.rhconecta.dev.domain.home.entity.Badge;
import com.coppel.rhconecta.dev.domain.home.entity.Banner;

import java.util.List;
import java.util.Map;

/**
 *
 */
public interface HomeRepository {

    /**
     *
     */
    void getBanners(
            UseCase.OnResultFunction<Either<Failure, List<Banner>>> callback
    );

    /**
     *
     */
    void getBadges(
            UseCase.OnResultFunction<Either<Failure, Map<Badge.Type, Badge>>> callback
    );

    /**
     * Gets data for help desk availability.
     */
    void getHelpDeskServiceAvailability(
            UseCase.OnResultFunction<Either<Failure, ZendeskResponse>> callback
    );

}

File: app/src/main/java/com/coppel/rhconecta/dev/data/home/HomeRepository.kt
package com.coppel.rhconecta.dev.data.home

import com.coppel.rhconecta.dev.domain.common.Either
import com.coppel.rhconecta.dev.domain.common.UseCase.OnResultFunction
import com.coppel.rhconecta.dev.domain.common.failure.Failure
import com.coppel.rhconecta.dev.domain.home.entity.Badge
import com.coppel.rhconecta.dev.domain.home.entity.Banner
import com.coppel.rhconecta.dev.domain.home.entity.HelpDeskAvailability

/**
 *
 */
interface HomeRepository {
    /**
     *
     */
    fun getBanners(
        callback: OnResultFunction<Either<Failure, List<Banner>>>,
    )

    /**
     *
     */
    fun getBadges(
        callback: OnResultFunction<Either<Failure, Map<Badge.Type, Badge>>>,
    )

    /**
     * Gets data for help desk availability.
     */
    suspend fun getHelpDeskServiceAvailability(): Either<Failure, HelpDeskAvailability>
}